examples/sql_bank/main.go

```diff
@@ -0,0 +1,189 @@
+
+func moveMoney(db *sql.DB) {
+	for {
+		from, to := rand.Intn(numAccounts), rand.Intn(numAccounts)
+		if from == to {
+			continue
+		}
+		amount := rand.Intn(maxTransfer)
+		tx, err := db.Begin()
+		if err != nil {
+			log.Fatal(err)
+		}
+		if _, err = tx.Exec("SET TRANSACTION ISOLATION LEVEL SERIALIZABLE"); err != nil {
+			log.Fatal(err)
+		}
+		rows, err := tx.Query("SELECT id, balance FROM accounts WHERE id IN ($1, $2)", from, to)
+		if err != nil {
+			log.Fatal(err)
+		}
+		var fromBalance, toBalance int
+		for rows.Next() {
+			var id, balance int
+			if err = rows.Scan(&id, &balance); err != nil {
+				log.Fatal(err)
+			}
+			switch id {
+			case from:
+				fromBalance = balance
+			case to:
+				toBalance = balance
+			default:
+				panic(fmt.Sprintf("got unexpected account %d", id))
+			}
+		}
+
+		if _, err = tx.Exec("UPDATE accounts SET balance=$2 WHERE id=$1", from, fromBalance-amount); err != nil {
+			if log.V(1) {
+				log.Warning(err)
+			}
+			if err = tx.Rollback(); err != nil {
+				log.Fatal(err)
+			}
+			continue
+		}
+		if _, err = tx.Exec("UPDATE accounts SET balance=$2 WHERE id=$1", to, toBalance+amount); err != nil {
+			if log.V(1) {
+				log.Warning(err)
+			}
+			if err = tx.Rollback(); err != nil {
+				log.Fatal(err)
+			}
+			continue
+		}
+		if err = tx.Commit(); err != nil {
+			if log.V(1) {
+				log.Warning(err)
+			}
+		} else {
+			atomic.AddUint64(&numTransfers, 1)
+		}
+	}
+}
+
+func verifyBank(db *sql.DB) {
+	var sum int64
+
+	if aggregate {
+		if err := db.QueryRow("SELECT SUM(balance) FROM accounts").Scan(&sum); err != nil {
+			log.Fatal(err)
+		}
+	} else {
+		tx, err := db.Begin()
+		if err != nil {
+			log.Fatal(err)
+		}
+		rows, err := tx.Query("SELECT balance FROM accounts")
+		if err != nil {
+			log.Fatal(err)
+		}
+		for rows.Next() {
+			var balance int64
+			if err = rows.Scan(&balance); err != nil {
+				log.Fatal(err)
+			}
+			sum += balance
+		}
+		if err = tx.Commit(); err != nil {
+			log.Fatal(err)
+		}
+	}
+
+	if sum == 0 {
+		log.Info("The bank is in good order.")
+	} else {
+		log.Fatalf("The bank is not in good order. Total value: %d", sum)
+	}
+}
+
+func main() {
+	db, err := sql.Open("postgres", "dbname=postgres sslmode=disable")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	if _, err = db.Exec("CREATE DATABASE bank"); err != nil {
+		if pqErr, ok := err.(*pq.Error); !ok || pqErr.Code.Name() != "duplicate_database" {
+			log.Fatal(err)
+		}
+	}
+
+	db, err = sql.Open("postgres", "dbname=bank sslmode=disable")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	db.SetMaxOpenConns(concurrency)
+
+	if _, err = db.Exec("CREATE TABLE IF NOT EXISTS accounts (id BIGINT UNIQUE NOT NULL, balance BIGINT NOT NULL)"); err != nil {
+		log.Fatal(err)
+	}
+
+	if _, err = db.Exec("TRUNCATE TABLE accounts"); err != nil {
+		log.Fatal(err)
+	}
+
+	for i := 0; i < numAccounts; i++ {
+		if _, err = db.Exec("INSERT INTO accounts (id, balance) VALUES ($1, $2)", i, 0); err != nil {
```

Comment:
This could be a single INSERT statement using `VALUES (a, b), (c, d), (e, f), ...`.